rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/widget/CovidRDTBarcodeFactory.java

```diff
@@ -63,6 +76,37 @@ public void onActivityResult(int requestCode, int resultCode, Intent data) {
 
     protected abstract String[] splitCSV(String barcodeCSV);
 
+    protected void populateBarcodeData(Intent data) throws JSONException {
+
+        JSONObject jsonObject = new JSONObject(data.getStringExtra("data"));
+
+        formUtils.getNextUniqueIds(null, (args, uniqueIds) -> {
+            List<String> ids = new ArrayList<>();
+            for (UniqueId uniqueId : uniqueIds) {
+                String currUniqueId = uniqueId.getOpenmrsId();
+                if (StringUtils.isNotBlank(currUniqueId)) {
+                    currUniqueId = currUniqueId.replace("-", "");
+                    ids.add(currUniqueId);
+                }
+            }
+
+            String uniqueId = Utils.isEmptyCollection(ids) ? "" : ids.get(0);
+
+            try {
+                jsonObject.put(BATCH_ID, uniqueId);
+                JsonApi jsonApi = (JsonApi) widgetArgs.getContext();
+                String stepName = widgetArgs.getStepName();
+
+                jsonApi.writeValue(stepName, CovidConstants.FormFields.QR_CODE_READER, jsonObject.toString(),  "", "", "", false);
+                jsonApi.writeValue("step5", BATCH_ID, uniqueId,  "", "", "", false);
```

Comment:
`step5` should ideally not be hard-coded. Could we move this to the step state config?

Reply:
fixed.